src/inmanta/module.py

```diff
@@ -1357,23 +1409,46 @@ def get_all_submodules(self) -> List[str]:
 
         return modules
 
+    @abstractmethod
+    def _get_plugin_dir(self) -> Optional[str]:
+        """
+        Return directory containing the python files which define handlers and plugins.
+        If no such directory is defined, this method returns None.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def _get_model_dir(self) -> str:
+        """
+        Return a path to the model directory of the module.
+        """
+        raise NotImplementedError()
+
+    # This method is not part of core's stable API but it is currently used by pytest-inmanta (inmanta/pytest-inmanta#76)
+    def _get_fq_mod_name_for_py_file(self, py_file) -> str:
+        """
+        Returns the fully qualified Python module name for a path to a python file in the plugins directory.
+
+        :param py_file: Absolute path to the python file in the plugins directory of the module.
+        """
+        rel_py_file = os.path.relpath(py_file, start=self._get_plugin_dir())
+        return loader.PluginModuleLoader.convert_relative_path_to_module(self.name, rel_py_file)
+
     def get_plugin_files(self) -> Iterator[Tuple[Path, ModuleName]]:
```

Comment:
I don't understand this suggestion. What do you want to move exactly?

Reply:
The `get_plugin_files` method. I thought you made changes to the implementation specifically to allow for v2 modules (the introduction of `_get_plugin_dir()` in particular), which wouldn't be required since this method should never be called for V2. But `_get_plugin_dir()` is also used by other methods, so my comment loses weight. I still think it could be moved to V1 but I don't think it matters that much.

Reply:
I don't see why `get_plugin_files()` would never be called for V2 modules?

Reply:
I'd was thinking the only reason this method exists is to find the plugin files that need to be loaded using the custom loader. But now that I think about it the `load_plugins` call is meant to load all plugin functions and register them to the `Namespace`. In that case we still need to load them for v2. In short, I was wrong, this method should definitely stay in the generic `Module` class.